client/src/Chat.jsx

```diff
@@ -1,26 +1,34 @@
-import React from "react";
+/* eslint-disable react/display-name */
+/* eslint-disable react/prop-types */
+import React from 'react';
 
 import {
   ApolloClient,
   InMemoryCache,
   ApolloProvider,
-  useQuery,
+  useSubscription,
   gql,
-  useMutation,
-} from "@apollo/client";
+  useMutation
+} from '@apollo/client';
 
-import dotenv from "dotenv";
-dotenv.config();
+import { WebSocketLink } from '@apollo/client/link/ws';
+import { Container, Row, Col, FormInput, Button } from 'shards-react';
 
-import { Container, Row, Col, FormInput, Button } from "shards-react";
+const link = new WebSocketLink({
+  uri: `ws://${process.env.websockerUrl}`,
```

Comment:
I recommend you to use `||` operator here to check if env variable is present. And if not, you may use some default value for local development, for example `localhost`. The same about other environment variables.

Reply:
Done